internal/olm/operator/registry/operator_installer.go

```diff
@@ -73,14 +74,20 @@ func (o OperatorInstaller) InstallOperator(ctx context.Context) (*v1alpha1.Clust
 		return nil, err
 	}
 
+	var subscription *v1alpha1.Subscription
 	// Create Subscription
-	if err = o.createSubscription(ctx, cs); err != nil {
+	if subscription, err = o.createSubscription(ctx, cs); err != nil {
+		return nil, err
+	}
+
+	// Wait for the Install Plan to be generated
+	if err = o.waitForInstallPlan(ctx, subscription); err != nil {
 		return nil, err
 	}
 
 	// Approve Install Plan (if necessary)
 	if approver, ok := o.CatalogCreator.(InstallPlanApprover); ok {
-		if err = approver.Approve(ctx, o.PackageName); err != nil {
+		if err = approver.Approve(ctx, subscription); err != nil {
 			return nil, err
 		}
 	}
```

Comment:
Why is the `CatalogCreator` be involved with approving install plans?

It doesn't seem like we need different implementations between run packagemanifests and run bundle. In both cases, we should create the subscription with manual approval and then immediately approve the install plan for that CSV.